presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -70,31 +66,42 @@ public static Builder builder()
 
     public static final class Builder
     {
-        private Estimate dataSize = unknownValue();
         private Estimate nullsFraction = unknownValue();
-        private Estimate distinctValuesCount = unknownValue();
+        private List<RangeColumnStatistics> rangeColumnStatistics = new ArrayList<>();
 
-        public Builder setDataSize(Estimate dataSize)
+        public Builder setNullsFraction(Estimate nullsFraction)
         {
-            this.dataSize = requireNonNull(dataSize, "dataSize can not be null");
+            this.nullsFraction = nullsFraction;
             return this;
         }
 
-        public Builder setNullsFraction(Estimate nullsFraction)
+        public Builder addRange(Consumer<RangeColumnStatistics.Builder> rangeBuilderConsumer)
         {
-            this.nullsFraction = nullsFraction;
+            RangeColumnStatistics.Builder rangeBuilder = RangeColumnStatistics.builder();
+            rangeBuilderConsumer.accept(rangeBuilder);
+            addRange(rangeBuilder.build());
+            return this;
+        }
+
+        public Builder addRange(Object lowValue, Object highValue, Consumer<RangeColumnStatistics.Builder> rangeBuilderConsumer)
```

Comment:
If you want to add a convenience overload for `addRange(Consumer<RangeColumnStatistics.Builder>`, consider having it without `Consumer<RangeColumnStatistics.Builder>`